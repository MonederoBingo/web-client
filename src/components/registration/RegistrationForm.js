import React from 'react';
import TextInput from '../common/TextInput';
import PasswordInput from '../common/PasswordInput';

const RegistrationForm = ({user, onSave, onChange, onCancel, saving, errors}) => {
  let actionsStyle = {
    marginRight: 20,
    marginTop: 10
  };

  let submitStyle = {
    marginTop: 10,
    marginRight: 50
  };

  return (
    <form>
      <TextInput
        name="companyName"
        label="Company Name"
        placeholder="Company name"
        value=""
        onChange={() => {}}
        error=""/>

      <TextInput
        name="email"
        label="Email"
        placeholder="Email"
        value={user.email}
        onChange={onChange}
        error={errors.email}/>

      <PasswordInput
        name="password"
        label="Password"
        placeholder="Password"
        value={user.password}
        onChange={onChange}
        error={errors.password}/>

      <input
        type="submit"
        disabled={saving}
        value={saving ? 'Saving...' : 'Save'}
        className="btn btn-primary"
        onClick={onSave}
        style={submitStyle}/>

      <input
        id="cancel"
        type="submit"
        value="Cancel"
        className="btn btn-default"
        onClick={onCancel}
        style={actionsStyle}/>
    </form>
  );
};

export default RegistrationForm;
