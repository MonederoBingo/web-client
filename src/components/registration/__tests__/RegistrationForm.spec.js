import React from 'react';
import expect from 'jest-matchers';
import RegistrationForm from '../RegistrationForm';
import {shallow} from 'enzyme';
import NumberInput from "../../common/NumberInput";

describe("RegistrationForm", () => {
  const props = {
    user: {
      email: "",
      password: ""
    },
    errors: {
      email: "",
      password: ""
    },
    onChange: () => {}
  };
  it("should show initial layout", () => {
    // when
    const wrapper = shallow(<RegistrationForm {...props}/>);

    // then
    expect(wrapper.children().nodes).toMatchSnapshot();
  });
  it("companyName should call onChange", () => {
    // given
    let called = false;
    props.onChange = () => {
      called = true;
    };
    const input = shallow(<RegistrationForm {...props}/>).find("[name='companyName']");

    // when
    input.simulate('change');

    // then
    expect(called).toBe(true);
  });
  it("email should call onChange", () => {
    // given
    let called = false;
    props.onChange = () => {
      called = true;
    };
    const input = shallow(<RegistrationForm {...props}/>).find("[name='email']");

    // when
    input.simulate('change');

    // then
    expect(called).toBe(true);
  });
  it("password should call onChange", () => {
    // given
    let called = false;
    props.onChange = () => {
      called = true;
    };
    const input = shallow(<RegistrationForm {...props}/>).find("[name='password']");

    // when
    input.simulate('change');

    // then
    expect(called).toBe(true);
  });
  it("should call onSubmit when submit is clicked", () => {
    // given
    let called = false;
    props.onSave = () => { called = true; };
    const submit = shallow(<RegistrationForm {...props}/>).find("[id='submit']");

     // when
    submit.simulate('click');

     // then
     expect(called).toBe(true);
  });
  // it("should disable submit if saving param returns true", () => {
  //   // given
  //   props.saving = true;
  //
  //    // when
  //   const wrapper = shallow(<RegistrationForm {...props}/>).find("[id='save']");
  //
  //    // then
  //    expect(wrapper.prop('disabled')).toBe(true);
  // });
  // it("should enable submit if saving param returns false", () => {
  //   // given
  //   props.saving = false;
  //
  //    // when
  //   const wrapper = shallow(<RegistrationForm {...props}/>).find("[id='save']");
  //
  //    // then
  //    expect(wrapper.prop('disabled')).toBe(false);
  // });
  // it("should set phone number from props", () => {
  //   // given
  //   props.pointsAwarding.phoneNumber = "123";
  //
  //    // when
  //   const field = shallow(<RegistrationForm {...props}/>).find("[name='phoneNumber']");
  //
  //    // then
  //    expect(field.prop('value')).toBe("123");
  // });
  // it("should set amount from props", () => {
  //   // given
  //   props.pointsAwarding.amount = 123;
  //
  //    // when
  //   const field = shallow(<RegistrationForm {...props}/>).find("[name='amount']");
  //
  //    // then
  //    expect(field.prop('value')).toBe(123);
  // });
  // it("should set sale key from props", () => {
  //   // given
  //   props.pointsAwarding.saleKey = "key";
  //
  //    // when
  //   const field = shallow(<RegistrationForm {...props}/>).find("[name='saleKey']");
  //
  //    // then
  //    expect(field.prop('value')).toBe("key");
  // });
});
