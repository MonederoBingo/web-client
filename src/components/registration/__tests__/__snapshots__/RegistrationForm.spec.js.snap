// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationForm should show initial layout 1`] = `
Array [
  <TextInput
    error=""
    label="Company Name"
    name="companyName"
    onChange={[Function]}
    placeholder="Company name"
    value=""
  />,
  <TextInput
    error=""
    label="Email"
    name="email"
    onChange={[Function]}
    placeholder="Email"
    value=""
  />,
  <PasswordInput
    error=""
    label="Password"
    name="password"
    onChange={[Function]}
    placeholder="Password"
    value=""
  />,
  <input
    className="btn btn-primary"
    disabled={undefined}
    onClick={undefined}
    style={
      Object {
        "marginRight": 50,
        "marginTop": 10,
      }
    }
    type="submit"
    value="Save"
  />,
  <input
    className="btn btn-default"
    id="cancel"
    onClick={undefined}
    style={
      Object {
        "marginRight": 20,
        "marginTop": 10,
      }
    }
    type="submit"
    value="Cancel"
  />,
]
`;
